packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -2796,6 +2804,16 @@ function updateDehydratedSuspenseComponent(
             attemptHydrationAtLane,
             eventTime,
           );
+
+          // Throw a special object that signals to the work loop that it should
+          // interrupt the current render.
+          //
+          // Because we're inside a React-only execution stack, we don't
+          // strictly need to throw here — we could instead modify some internal
+          // work loop state. But using an exception means we don't need to
+          // check for this case on every iteration of the work loop. So doing
+          // it this way moves the check out of the fast path.
+          throw SelectiveHydrationException;
         } else {
```

Comment:
Is this branch just a bug in React? Not sure how it happens exactly.

Reply:
Same answer as here: https://github.com/facebook/react/pull/25695/commits/cd34ec31e96f765a342b0f60b8930274f64bbe32#r1024411227

Reply:
This link is broken.

Reply:
It happens when the hydration lane is suspended. That indicates we already attempted to hydrate but we couldn't because the code/data isn't ready. In that case we switch back to the original update lane and render without hydrating. Without this branch it would keep restarting in a loop until the data resolved.